### PR TITLE
Sidecar tagged assignments

### PR DESCRIPTION
**Pull request by mpfz0r**

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/plugins/sidecar/services/EtagService.java [hunk 1]
```diff
@@ -62,41 +88,82 @@ public EtagService(SidecarPluginConfiguration pluginConfiguration,
...
-        if (event.etag().equals("")) {
-            LOG.trace("Invalidating all collector configuration etags");
+        //noinspection rawtypes
+        Cache cache;
+        switch (event.cacheContext()) {
+            case COLLECTOR -> cache = collectorCache;
+            case CONFIGURATION -> cache = configurationCache;
+            case REGISTRATION -> cache = registrationCache;
+            default -> throw new IllegalArgumentException("Unknown cache context");
+        }
```
Comment (thll) [inline graylog2-server/src/main/java/org/graylog/plugins/sidecar/services/EtagService.java:98]: Playing around with the new java language features, sorry 😬 . What do you think about this simplification?

```suggestion
        var cache = switch (event.cacheContext()) {
            case COLLECTOR -> collectorCache;
            case CONFIGURATION -> configurationCache;
            case REGISTRATION -> registrationCache;
        };
```

Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog/plugins/sidecar/services/EtagService.java:98]: sure, why not :+1:

File: graylog2-server/src/main/java/org/graylog2/contentpacks/facades/SidecarCollectorConfigurationFacade.java [hunk 2]
```diff
@@ -91,11 +91,12 @@ public NativeEntity<Configuration> createNativeEntity(Entity entity,
...
     private NativeEntity<Configuration> decode(EntityV1 entity, Map<String, ValueReference> parameters) {
         final SidecarCollectorConfigurationEntity configurationEntity = objectMapper.convertValue(entity.data(), SidecarCollectorConfigurationEntity.class);
-        final Configuration configuration = Configuration.create(
+        final Configuration configuration = Configuration.createWithoutId(
                 configurationEntity.collectorId().asString(parameters),
                 configurationEntity.title().asString(parameters),
                 configurationEntity.color().asString(parameters),
-                configurationEntity.template().asString(parameters));
+                configurationEntity.template().asString(parameters),
+                Set.of());
```
Comment (thll) [inline graylog2-server/src/main/java/org/graylog2/contentpacks/facades/SidecarCollectorConfigurationFacade.java:99]: I guess we need to support tags in content packs as well. We can do that in another PR though.

Comment (mpfz0r) [reply graylog2-server/src/main/java/org/graylog2/contentpacks/facades/SidecarCollectorConfigurationFacade.java:99]: I thought about that, but tags might be very specific to your environment, so I think we can live without them..

Comment (thll) [reply graylog2-server/src/main/java/org/graylog2/contentpacks/facades/SidecarCollectorConfigurationFacade.java:99]: I like your answer 👍 😄

**Review by thll — APPROVED** [review]
Looks good. Testing will be easier with the UI present, but we can already merge this PR into the feature branch. I'll add some TODOs to https://github.com/Graylog2/graylog-plugin-enterprise/issues/3956.
